src/components/dialogs/export-dialog.js

```diff
@@ -4,20 +4,101 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-import { Dialog, DialogTitle, Grid } from '@mui/material';
+import {
+    Collapse,
+    Dialog,
+    DialogTitle,
+    List,
+    ListItem,
+    ListItemText,
+    Switch,
+    TextField,
+    Tooltip,
+    Typography,
+} from '@mui/material';
+import ExpandLessIcon from '@mui/icons-material/ExpandLess';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
 import { FormattedMessage, useIntl } from 'react-intl';
 import DialogContent from '@mui/material/DialogContent';
 import DialogActions from '@mui/material/DialogActions';
 import Button from '@mui/material/Button';
 import PropTypes from 'prop-types';
-import React, { useEffect } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import InputLabel from '@mui/material/InputLabel';
 import Alert from '@mui/material/Alert';
 import FormControl from '@mui/material/FormControl';
 import Select from '@mui/material/Select';
 import MenuItem from '@mui/material/MenuItem';
 import CircularProgress from '@mui/material/CircularProgress';
 import { getAvailableExportFormats, getExportUrl } from '../../utils/rest-api';
+import IconButton from '@mui/material/IconButton';
+
+function longestCommonPrefix(strs) {
+    if (!strs?.length) return '';
+    let prefix = strs.reduce((acc, str) =>
+        str.length < acc.length ? str : acc
+    );
+
+    for (let str of strs) {
+        while (str.slice(0, prefix.length) !== prefix) {
+            prefix = prefix.slice(0, -1);
+        }
+    }
+    return prefix;
+}
+
+const useMeta = (metasAsArray) => {
+    const longestPrefix = longestCommonPrefix(metasAsArray.map((m) => m.name));
+    const lastDotIndex = longestPrefix.lastIndexOf('.');
+    const prefix = longestPrefix.slice(0, lastDotIndex + 1);
+
+    const defaultInst = useMemo(() => {
+        return Object.fromEntries(
+            metasAsArray.map((m) => {
+                if (m.type === 'BOOLEAN') return [m.name, m.defaultValue];
+                return [m.name, m.defaultValue ?? null];
+            })
+        );
+    }, [metasAsArray]);
+    const [inst, setInst] = useState(defaultInst);
```

Comment:
What 'inst' means?

Reply:
inst is for 'instance', data at level 1. Meta is kinda class, describes data at level 1. Pretty practical to use those two 4 letters identifier bases that are often handled with resembling code.